bots/hgbridge/src/main/java/org/openjdk/skara/bots/hgbridge/ExporterConfig.java

```diff
@@ -168,7 +168,8 @@ void sponsors(List<String> sponsors) {
     }
 
     public Converter resolve(Path scratchPath) throws IOException {
-        var localRepo = Repository.materialize(scratchPath, configurationRepo.url(), configurationRef);
+        var localRepo = Repository.materialize(scratchPath, configurationRepo.url(),
+                                               configurationRef + ":hgbridge_config_" + configurationRepo.name());
```

Comment:
Why no `+` here to to imply `--force` to `git fech`?

Reply:
No reason, good catch!

Reply:
Should we instead change `Repository.materialize` to always prepend the `+` to the refspec? Or do you think that makes it less clear at the call sites?

Reply:
Not sure, it's something you only want if you have a target ref, and only if you are using git.. And since materialize is implemented as a default method in Repository, it might be a little odd to have that logic there. Perhaps it could be refactored to not be a default method though.

Reply:
Ah right. No, `Repository.materialize` is `static`, not `default` :wink: I think we should leave as it is then.

The only other approach I can think if something like `Repository.materalize(Path path, URI remote, String fromRef, String toRef, boolean force, boolean checkout`), but then the parameter list is becoming really long (and `hg` can't handle `fromRef != toRef` anyhow). So I think we just leave it as it is.